Document Rekordbox library import (#322)
mixxxdj/manual: source/chapters/library.rst

  Mac partition mounted in Linux, you can load it and use your iTunes tracks and
  playlists as well.

.. _library-rekordbox:

Using the Rekordbox library
---------------------------
To read Rekordbox databases for CDJ/XDJ players on USB or SD devices,
the database has to be exported to a USB or SD device with a FAT or HFS
file system using the Rekordbox Export mode. Instructions on how to do this can be found in this `article by Pioneer DJ <https://forums.pioneerdj.com/hc/en-us/articles/205021549-Moving-your-rekordbox-database-Video->`__ .

> Databases that have been moved to an external device via Preferences > Advanced > Database management are not supported.

Reviewer: ```suggestion
Databases that have been moved to an external device via :guilabel:`Preferences > Advanced > Database management` in Rekordbox are not supported.
```

Reviewer: Make it clear that menu is in Rekordbox, not Mixxx.

The following data is read:

* Folders
* Playlists
* Beatgrids
* Hot cues
* Memory cues

> * Loops (not yet supported in Mixxx 2.3)

Reviewer: I thought only the first loop was supported in 2.3?

Reviewer: Yes, I just checked [the blog post about this](https://mixxx.org/news/2020-07-20-new-in-2-3-rekordbox-support/). Also, the manual should explain how Rekordbox's memory and hotcues are imported to Mixxx's hotcues.

Author: I just looked this up. You're right.  Since Mixxx currently has one loop, during importation, only the first chronological Rekordbox loop is mapped to the Mixxx loop. The rest of the loops are appended as Mixxx hotcues following the previously imported Rekordbox hotcues.

Author: > Yes, I just checked [the blog post about this](https://mixxx.org/news/2020-07-20-new-in-2-3-rekordbox-support/). Also, the manual should explain how Rekordbox's memory and hotcues are imported to Mixxx's hotcues.

Alright. Let me add this.

Reviewer 2: I think the is paragraph should be merged with the "How Rekordbox X is Imported".

> 1. Plug your USB or SD device in while Mixxx is running.

Reviewer: ```suggestion
1. Plug your USB or SD drive in while Mixxx is running.
```

> 2. Click on Rekordbox in the library and all USB drives prepared with Rekordbox will be visible. If this does not happen right away, then you can click on 'Check for attached Rekordbox devices (refresh)'.

Reviewer: ```suggestion
2. Click on Rekordbox in the library and all USB drives prepared with Rekordbox will be visible. If this does not happen right away, then you can click on :guilabel:`Check for attached Rekordbox devices (refresh)`.
```

> 3. Click on the desired removable device, and all your folders and playlists will be revealed for you to play from.

Reviewer: ```suggestion
3. Click on the desired removable device. All your folders and playlists will be revealed for you to play from.
```

  playlists as well.

.. _library-rekordbox:

Using the Rekordbox library
---------------------------
To read Rekordbox databases for CDJ/XDJ players on USB or SD devices,
the database has to be exported to a USB or SD device with a FAT or HFS
file system using the Rekordbox Export mode. Instructions on how to do this can be found in this `article by Pioneer DJ <https://forums.pioneerdj.com/hc/en-us/articles/205021549-Moving-your-rekordbox-database-Video->`__ .
Databases that have been moved to an external device via :guilabel:`Preferences > Advanced > Database management` in Rekordbox are not supported.

The following data is read:

* Folders
* Playlists
* Beatgrids
* Hot cues
* Memory cues
* Loops (not yet supported in Mixxx 2.3)

1. Plug your USB or SD drive in while Mixxx is running.
2. Click on Rekordbox in the library and all USB drives prepared with Rekordbox will be visible. If this does not happen right away, then you can click on :guilabel:`Check for attached Rekordbox devices (refresh)`.
3. Click on the desired removable device. All your folders and playlists will be revealed for you to play from.

> **How Rekordbox hotcues, memory cues, and loops are imported**

Reviewer 2: Please use a regular headline. Right now we are often using using bold text for headlines, which makes it impossible to link to.

Author: Okay. .. Let me make a commit for this so you review the headline style that I've used.

  Mac partition mounted in Linux, you can load it and use your iTunes tracks and
  playlists as well.

.. _library-rekordbox:

Using the Rekordbox library
---------------------------
To read Rekordbox databases for CDJ/XDJ players on USB or SD devices,
the database has to be exported to a USB or SD device with a FAT or HFS

> file system using the Rekordbox Export mode. Instructions on how to do this can be found in this `article by Pioneer DJ <https://forums.pioneerdj.com/hc/en-us/articles/205021549-Moving-your-rekordbox-database-Video->`__ .

Reviewer 2: I don't think this is the right article. It's about moving your DB rather and USB export. However, the Pioneer documentation is pretty bad and I can t find a better link. Maybe we should just remove the last sentence.

Author: I failed to find proper Pioneer DJ documentation too. Let me remove  it.

Databases that have been moved to an external device via :guilabel:`Preferences > Advanced > Database management` in Rekordbox are not supported.

1. Plug your USB or SD drive in while Mixxx is running.

> 2. Click on Rekordbox in the library and all Rekordbox prepared USB drives will be visible. If this does not happen right away, then you can click on :guilabel:`Check for attached Rekordbox devices (refresh)`.

Reviewer 2: ```suggestion
2. Click on the :guilabel:`Rekordbox` item in the library sidebar to scan for  Rekordbox prepared USB drives. If no Rekordbox devices show up, you can click on :guilabel:`Check for attached Rekordbox devices (refresh)`.
```

3. Click on the desired removable device. All your folders and playlists will be revealed for you to play from.

How Rekordbox hotcues, memory cues, and loops are imported
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

>   The following data is read:

Reviewer 2: ```suggestion
The following data is read:
```

> ----------------------------------------------------------

Reviewer 2: This section is now at the same level as the "Using the Rekordbox library", but it should be a subsection of it: https://docutils.sourceforge.io/docs/user/rst/quickstart.html#sections

  * Folders
  * Playlists
  * Beatgrids
  * Hot cues
  * Memory cues
  * Loops

>   Mixxx currently has one main cue point, one loop, and thirty eight hotcues. As such:

Reviewer 2: ```suggestion
Mixxx currently has one main cue point, one loop, and thirty eight hotcues. As such:
```

  *  All Rekordbox hotcues are mapped to Mixxx hotcues as expected, the first chronological Rekordbox memory cue is mapped to the Mixxx main cue.
  * The first chronological Rekordbox loop is also mapped to the Mixxx loop.
  * All loops (including the first loop), and all subsequent Rekordbox memory cues are appended as Mixxx hotcues following the previously imported Rekordbox hotcues.
  * The hotcue and memory cue colors you assigned in Rekordbox are also imported, assisting in distinguishing between them.

> Whilst the additional imported Rekordbox loops only appear in Mixxx 2.3 as hotcues, their loop details are preserved, ready for the multiple loop feature proposed for Mixxx 2.4.

Reviewer 2: ```suggestion
Although additional imported Rekordbox loops only appear in Mixxx 2.3 as hotcues, their loop information is stored in the Mixxx library will become usable in Mixxx 2.4.
```

> Although additional imported Rekordbox loops only appear in Mixxx 2.3 as hotcues, their loop information is stored in the Mixxx library will become usable in Mixxx 2.4.

Reviewer 2: ```suggestion
Although additional imported Rekordbox loops only appear in Mixxx 2.3 as hotcues, their loop information is stored in the Mixxx library and will become usable in Mixxx 2.4.
```
